app/serializers/api/v2/csv/csv_serializer.rb

```diff
@@ -0,0 +1,56 @@
+module Api
+  module V2
+    module Csv
+      module CsvSerializer
+        extend ActiveSupport::Concern
+
+        def initialize(serializables, options = {})
+          @serializables = serializables
+          @options = options
+        end
+
+        def serializable_array
+          @serializables.each_with_object(initial_rows) do |serializable, acc|
+            data_row = columns.map do |column|
+              serializable.public_send(column)
+            end
+
+            acc << data_row
+          end
+        end
+
+        def serialized_csv
+          serialized = CSV.generate do |csv|
+            serializable_array.each do |row|
```

Comment:
This will build up a duplicate row array (of column arrays) of all data to be exported?

```ruby
def serialized_csv
  CSV.generate do |csv|
    add_header_row(csv)

    @serializables.each do |serializable|
      add_data_row(csv, serializable)
    end
  end
end

def add_data_row(csv, serializable)
  csv << columns.map do |column|
    serializable.public_send(column)
  end
end
```

Reply:
I'm not seeing any duplicates?

Reply:
What I've done achieves this already, tbh